lib/sql.ts

```diff
@@ -259,12 +263,154 @@ export class SQLDatabaseManager extends QueryProcessManager<DatabaseQuery, any>
 	}
 }
 
+export const tables = new Map<string, DatabaseTable<any>>();
+
+export class DatabaseTable<T> {
+	database: SQLDatabaseManager;
+	name: string;
+	primaryKeyName: string;
+	constructor(
+		name: string,
+		primaryKeyName: string,
+		database: SQLDatabaseManager
+	) {
+		this.name = name;
+		this.database = database;
+		this.primaryKeyName = primaryKeyName;
+		tables.set(this.name, this);
+	}
+	private SQL: typeof import('sql-template-strings').SQL = (() => {
+		try {
+			return require('sql-template-strings');
+		} catch {
+			return () => {
+				throw new Error("Using SQL-template-strings without it installed");
+			};
+		}
+	})();
+	async selectOne(
+		entries: string | string[],
+		where?: SQLStatement
+	): Promise<T | null> {
+		const query = where || this.SQL``;
+		query.append(' LIMIT 1');
+		const rows = await this.selectAll(entries, query);
+		return rows?.[0] || null;
+	}
+	selectAll(
+		entries: string | string[],
+		where?: SQLStatement
+	): Promise<T[]> {
+		const query = this.SQL`SELECT `;
+		if (typeof entries === 'string') {
```

Comment:
Errrr, should `table.selectAll('my_column');` return all values of `my_column` instead of `SELECT *`?
I guess you can do `table.selectAll(['my_column'])`?

Reply:
Yes, you can.

Reply:
Note that doing `selectAll('my_column')` is the same as `selectAll(['my_column'])`.

Reply:
> Note that doing `selectAll('my_column')` is the same as `selectAll(['my_column'])`.

The code you have appears to return values for all columns (`query.append(' * ')`) if the `entries` argument is a string, whereas when it's an array it only returns values for the specified columns. Am I missing something?

Reply:
You resolved this but didn't actually explain my question...

Reply:
it was fixed